lib/fika/type_checker.ex

```diff
@@ -286,6 +309,12 @@ defmodule Fika.TypeChecker do
     end)
   end
 
+  defp infer_map_key_value(env, {key, value}) do
+    {:ok, key_type, env} = infer_exp(env, key)
+    {:ok, val_type, env} = infer_exp(env, value)
+    {:ok, "Map(#{key_type},#{val_type})", env}
```

Comment:
This function isn't safe because it may raise a MatchError. infer_exp can return the `{:error, reason}` tuple.

Reply:
I think something like this should do:

```elixir
with {:ok, key_type, env} <- infer_exp(env, key),
        {:ok, value_type, env} <- infer_exp(env, value) do
  Enum.reduce_while(rest, ...ensure keys and values match the types...)
end
```

or some other way you think is better, but always remember to handle {:error, reason} tuple when inferring type of any expression.